knowledge_storm/rm.py

```diff
@@ -396,11 +426,130 @@ def forward(self, query_or_queries: Union[str, List[str]], exclude_urls: List[st
             related_docs = self.qdrant.similarity_search_with_score(query, k=self.k)
             for i in range(len(related_docs)):
                 doc = related_docs[i][0]
-                collected_results.append({
-                    'description': doc.metadata['description'],
-                    'snippets': [doc.page_content],
-                    'title': doc.metadata['title'],
-                    'url': doc.metadata['url'],
-                })
+                collected_results.append(
+                    {
+                        "description": doc.metadata["description"],
+                        "snippets": [doc.page_content],
+                        "title": doc.metadata["title"],
+                        "url": doc.metadata["url"],
+                    }
+                )
+
+        return collected_results
+
+
+class SerperRM(dspy.Retrieve):
+    """Retrieve information from custom queries using Serper.dev.
+
+    To be compatible with STORM, the results should have the following fields:
+        - snippet: Snippets that will be used for the document
+        - title: The title of the document.
+        - url: The URL of the document. STORM use url as the unique identifier of the document, so ensure different
+            documents have different urls.
+        - description (optional): The description of the document.
+
+    """
+
+    def __init__(self, serper_search_api_key=None, query_params=None):
+        super().__init__()
+        self.usage = 0
+        self.query_params = query_params
+        self.serper_search_api_key = serper_search_api_key
+        if not self.serper_search_api_key and not os.environ.get("SERPER_API_KEY"):
```

Comment:
Could you add comment for `__init__` to include where to obtain the `SERPER_API_KEY` and information about `query_params`?